src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userByEmailOptional = userService.findByEmail(email);
+        if (userByEmailOptional.isPresent()
+                && userByEmailOptional.get().getPassword()
+                .equals(HashUtil.hashPassword(password, userByEmailOptional.get().getSalt()))) {
+            return userByEmailOptional.get();
+        }
+        throw new AuthenticationException("Can't authenticate user");
+    }
+
+    @Override
+    public User register(String email, String password) throws AuthenticationException {
+        if (userService.findByEmail(email).isEmpty()) {
+            return userService.add(new User(email, password));
+        }
+        throw new AuthenticationException("User with such email: " + email
```

Comment:
It seems to me that "AuthenticationException"  is not suitable name for registration. It might be worth creating a separate exception?